File: App/Resources/AlgorithmDetails/burntpancakesort.bundle/burntpancakesort.py
def flip(arr, end):
  start = 0
  while start < end:
    arr[start], arr[end] = arr[end], arr[start]
    start += 1
    end -= 1

def sort(arr):
  n = len(arr)
  for i in range(n - 1, 0, -1):
    max_index = 0
    for j in range(max_index + 1, i + 1):
      if arr[j] > arr[max_index]:
        max_index = j
    if max_index != i:
      flip(arr, max_index)
      flip(arr, i)
      flip(arr, i - 1)
      flip(arr, max_index - 1)

if __name__ == "__main__":
  array = [0, 39, 21, 62, 91, 77, 14, 23,
    90, 69, 51, 81, 68, 83, 32, 56]
  sort(array)
  print(array)

File: App/Resources/AlgorithmDetails/burntpancakesort/burntpancakesort.cpp
#include <cstdio>
#include <utility>

int array[16] = {0, 39, 21, 62, 91, 77, 14, 23,
                 90, 69, 51, 81, 68, 83, 32, 56};

void printList(int items[], int size) {
  for (int i = 0; i < size; i++) {
    if (i == 0) {
      printf("[%d, ", items[i]);
    } else if (i != size - 1) {
      printf("%d, ", items[i]);
    } else {
      printf("%d]", items[i]);
    }
  }
}

void flip(int arr[], int end) {
  int start = 0;
  while (start < end) {
    std::swap(arr[start], arr[end]);
    start++;
    end--;
  }
}

void sort(int arr[], int n) {
  for (int i = n - 1; i > 0; i--) {
    int max = 0;
    for (int j = max + 1; j <= i; j++) {
      if (arr[j] > arr[max]) {
        max = j;
      }
    }
    if (max != i) {
      flip(arr, max);
      flip(arr, i);
      flip(arr, i - 1);
      flip(arr, max - 1);
    }
  }
}

int main(int argc, char *argv[]) {
  int size = sizeof(array) / sizeof(array[0]);
  sort(array, size);
  printList(array, size);
  return 0;
}